src/react/ISV/hooks/useIAMUser.tsx

```diff
@@ -0,0 +1,103 @@
+import { useMutation, useQuery } from 'react-query';
+import { useMemo, useState } from 'react';
+import { useIAMClient } from '../../IAMProvider';
+import { useAssumeRoleQuery } from '../../next-architecture/ui/S3ClientProvider';
+
+interface IAMUser {
+  id: string;
+  name: string;
+}
+
+interface UseIAMUserProps {
+  IAMUserName: string;
+  IAMUserNameType: 'create' | 'existing';
+  onIAMUsersLoaded?: (users: IAMUser[]) => void;
+  onShouldGenerateKey?: (shouldGenerate: boolean) => void;
+}
+
+export const useIAMUser = ({
```

Comment:
I think we should review this hook all together @ziyang-lin-404 @JeanMarcMilletScality